### ISSUE TITLE
Desplazamiento cambio de sistema de coordenadas

### ISSUE THREAD

**Issue by nimonro** [neslerel/BetaTesters_QGRed#49]
Otro problema que tengo y para el que sería útil contar una herramienta es el siguiente. Los 30 modelos que tengo están en un sistema de coordenadas que no identifico. Entre ellos son coherentes pero no soy capaz de ubicarlos sobre el modelo de la red arterial de la que en la realidad se abastecen. Este último lo tengo en WGS 84 UTM 14N. 

He visto una herramienta de QGis que permite mover los elementos seleccionados de un shape (Mover Objetos Espaciales), pero de acuerdo con las pruebas que he hecho, no los de varios shapes al mismo tiempo. Entonces mover toda una red (tuberías, nudos, tanques, etc.) se dificulta bastante ya que mantener cada cosa en su lugar no es fácil.

Una herramienta para esto o modificar la existente para hacer estos movimientos sería muy interesante.

Gracias.

**Comment by neslerel** [comment]
Hola Nicolás!

Esto ya lo teníamos en mente hacer. Queremos crear una herramienta que te permita desplazar el modelo (ya veremos cómo el usuario lo define, si con coordenadas, distancia...) e incluso ajustar el modelo si éste está "deformado" al estar en otra proyección.

Espero que podamos desarrollarlo en un corto plazo.

Saludos.

**Comment by nimonro** [comment]
Perfecto, yo mientras tanto trataré de resolverlo ‘a mano’. Os cuento cómo me va.

Gracias.

**Comment by fmartine** [comment]
Efectivamente, EPANET-UI incluye esta opción en la versión para python, tal como se ve en la captura adjunta, con lo que estamos 'obligados' a incorporarla también a QGISRed. Puede ser muy útil para ubicar redes capturadas desde CAD, en las que se utilizó un sistema de referencia cualquiera, o para corregir  desplazamientos debidos al uso de sistemas de proyección distintos (siempre que sean proyecciones planares con poca distorsión).

![Captura_Coordenadas](https://user-images.githubusercontent.com/20404014/71697001-e35c5b80-2db6-11ea-8686-b80868376297.JPG)

EPANET pregunta por la correspondencia entre las coordenadas de 2 puntos, y opcionalmente permite especificar el sistema de proyección CRS  tras la transformación, las unidades de partida y corregir las longitudes de las tuberías si procede.

En QGISRed podemos ir un poco más allá, y asumir una transformación afin de hasta 5 parámetros: traslación en X, traslación en Y, distorsión lineal en X, distorsión lineal en Y, y Giro. En general, las distorsiones en X y Y deberían ser iguales (pues en una proy. geográfica 1 metro debe representarse mediante un segmento de la misma longitud en cualquier dirección), con lo que quedarían 4 parámetros.  Al fijar la correspondencia entre 2 puntos, se obtienen 4 ecuaciones que permiten determinar esos 4 parámetros.

En QGISRed podemos permitir declarar la correspondencia  entre más de 2 puntos, con lo que podríamos determinar los 5 parámetros, e incluso aplicar técnicas de regresión para determinar los parámetros que mejor resuelven la transformación minimizando el error cuadrático, cuando se dé el caso de un exceso de ecuaciones.

**Comment by nimonro** [comment]
Me has pillado con lo de Epanet-UI, ¿puedo descargarlo? ¿cómo corre? Por lo que veo y por el momento me puede ayudar a resolver mi problema de mover esta red ¿no?

Gracias.

**Comment by fmartine** [comment]
Hola Nicolás,

Hace unos años, la EPA sacó un concurso para desarrollar una nueva interface para EPANET y SWMM en código libre sobre plataformas más actuales. Lo ganó una empresa no esperada y el desarrollo va muy lento. Está sin acabar y apenas hay ya fondos. Este es el enlace para su descarga. Pulsa en Assets y te bajas el instalador que prefieras.
https://github.com/USEPA/SWMM-EPANET_User_Interface/releases

Como te digo, se cuelga cada dos por tres y tarda muchíiiiiiiiiiisimo en cargar una red grande. Prueba a ver si te va lo de desplazar una red. En cualquier caso es muy sencillo también realizar una transformacion afin con Excel. Cargas las secciones [COORDINATES] y [VERTEX] en Excel, aplicas la transformación y pegas de nuevo los resultados en el INP. ¿Recuerdas cómo hacíamos con las leyes de control cuando el modelo de Valencia del año 2000 ?. Pues algo parecido.

Finalmente la EPA ha optado también por pasar la interfaz de Delphi a la versión 10.0 y continuar mejorando dicha interfaz desde Delphi.  Al menos provisionalmente. En la página oficial de EPANET ya tienes una versión nueva de EPANET en Delphi 10 y pronto sacarán otra mejor, adaptada a la toolkit 2.2 recientemente liberada, donde podrás especificar desde la interfaz los  parámetros para las fugas y la demandas dependientes de la presión.

Las últimas novedades de EPANET en Delphi están en 
https://github.com/USEPA/EPANET2.2  pero no recuerdo cómo acceder al instalador

**Comment by nimonro** [comment]
Al final instalé el EPANET-UI y pude hacer el movimiento de la red con la herramienta que comentabas. Una vez desplazado veo que la red en algunos puntos encaja con el OpenStreetMap y en otros no, ahí ya no sabría decir si es una cuestión de los datos de origen (eso no los tengo) o de la traslación, aunque de momento para el propósito que quiero usar este modelo es suficiente.
Gracias!

**Comment by fmartine** [comment]
Me alegro que te haya servido. Cuando liberemos nuestra herramienta podrás ajustar todos los nodos que quieras y el resto se adaptarán respetando los nudos ajustados.
